lib/elixir/lib/map.ex

```diff
@@ -838,7 +828,11 @@ defmodule Map do
 
   """
   @spec equal?(map, map) :: boolean
+  def equal?(map1, map2)
+
   def equal?(%{} = map1, %{} = map2), do: map1 === map2
+  def equal?(map1, %{}), do: :erlang.error({:badmap, map1})
+  def equal?(%{}, map2), do: :erlang.error({:badmap, map2})
```

Comment:
What happens if both are not a map? The last case should probably not match on one of the args